recipes/sdl/3.x/conanfile.py

```diff
@@ -0,0 +1,193 @@
+                self.settings.os != "Android")
+
+    @property
+    def _supports_opengl(self):
+        # CMakeLists.txt#L297
+        # TODO: Add option for opengl
+        return (self.options.get_safe("video")
+                and self.settings.os not in ("iOS", "tvOS", "watchOS"))
+
+    @property
+    def _supports_libudev(self):
+        # CMakeLists.txt#L351&L1618
+        # TODO: Add option for libudev
+        return self.settings.os in ("Linux", "FreeBSD")
+
+    @property
+    def _supports_pulseaudio(self):
+        # CMakeLists.txt#L372
+        # TODO: Add option for pulseaudio
+        return self._is_unix_sys and self.options.get_safe("audio")
+
+    @property
+    def _supports_dbus(self):
+        # CMakeLists.txt#292
+        # TODO: Add option for dbus
+        return self._is_unix_sys
+
+    def requirements(self):
+        if self._needs_libusb:
+            self.requires("libusb/1.0.26")
+        if self._supports_opengl:
+            self.requires("opengl/system")
+        if self._supports_libudev:
+            self.requires("libudev/system")
+        if self._supports_dbus:
+            self.requires("dbus/1.15.8")
+        if self._supports_pulseaudio:
+            self.requires("pulseaudio/17.0")
+
+    def build_requirements(self):
+        self.tool_requires("cmake/[>=3.24 <4]")
+        if self.settings.os == "Linux" and not self.conf.get("tools.gnu:pkg_config", check_type=str):
+            self.tool_requires("pkgconf/[>=2.2 <3]")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["SDL_SHARED"] = self.options.shared
+        tc.variables["SDL_STATIC"] = not self.options.shared
+        # Todo: Remove after recipe develop is finished
+        tc.variables["SDL_TEST_LIBRARY"] = True
+        tc.variables["SDL_TESTS"] = True
+        tc.variables["SDL_EXAMPLES"] = True
+        tc.cache_variables["CMAKE_TRY_COMPILE_CONFIGURATION"] = str(self.settings.build_type)
+        if self._needs_libusb:
+            tc.variables["SDL_HIDAPI_LIBUSB"] = True
+            tc.variables["SDL_HIDAPI_LIBUSB_SHARED"] = self.dependencies["libusb"].options.get_safe("shared", False)
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.set_property("libusb", "cmake_target_name", "LibUSB::LibUSB")
+        deps.set_property("libusb", "cmake_additional_variables_prefixes", ["LibUSB"])
+        deps.generate()
+        pcdeps = PkgConfigDeps(self)
+        pcdeps.generate()
+
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "LICENSE.txt", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["SDL3"]
+        self.cpp_info.set_property("cmake_file_name", "SDL3")
+        self.cpp_info.set_property("cmake_target_name", "SDL3::SDL3")
```

Comment:
In addition, create a `SDL3::SDL3-static` when creating a static library and
`SDL3::SDL3-shared` when building a shared library.

Also create `SDL3::Headers` for the headers.